eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -13,7 +15,9 @@ jobs:
         displayName: "Use .NET Core sdk $(DotNetCoreSDKVersion)"
         inputs:
           version: "$(DotNetCoreSDKVersion)"
-      - script: "dotnet pack eng/service.proj -o $(Build.ArtifactStagingDirectory) -warnaserror /p:ServiceDirectory=${{parameters.ServiceDirectory}} /p:PublicSign=false ${{ parameters.VersioningProperties }}"
+      - script: >-
+          dotnet pack eng/service.proj -o $(Build.ArtifactStagingDirectory) -warnaserror /p:ServiceDirectory=${{parameters.ServiceDirectory}}
+          /p:PublicSign=false ${{ parameters.VersioningProperties }} /p:Configuration=$(BuildConfiguration)
```

Comment:
This is going to set Configuration to an invalid value for any cases that isn't Release. Take a look at the logs a little closer https://dev.azure.com/azure-sdk/public/_build/results?buildId=70188&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774. 

If you switch back to a template parameter I think you will get what you were trying for.